pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -149,12 +163,28 @@ func (r *ReconcilePolicy) removeFinalizerAndReleaseMac(virtualMachine *kubevirt.
 	}
 
 	// remove our finalizer from the list and update it.
-	log.V(1).Info("removing finalizers",
-		"virtualMachineName", request.Name,
-		"virtualMachineNamespace", request.Namespace)
-	virtualMachine.ObjectMeta.Finalizers = helper.RemoveString(virtualMachine.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName)
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Please consider extract this function to a new one

Reply:
please elaborate.

The reason we want to do Retry on conflict is that potentially, following vm delete, both kmp-mgr pods will try to remove their finalizers from the spec, which will cause conflict. This is ok, as long as they both succeed eventually.
Hopefully the DefaultRetry configuration will be sufficient , as it's supposed to be exactly the case it's supposed to handle.